src/utils/uint256.cairo

```diff
@@ -19,3 +33,39 @@ func uint256_exp{range_check_ptr}(a: Uint256, b: Uint256) -> Uint256 {
     let (res, _) = uint256_mul(a, pow);
     return res;
 }
+
+// @notice Performs sign extension on a Uint256 number.
+// @param x The number to be sign extended.
+// @param b The size in bytes minus one of x to consider.
+// @return The result of sign extending x from position b.
+// If b > 31, the result is x. Otherwise, the sign bit is extended from position b.
+func uint256_signextend{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
```

Comment:
Ok I've had to read both the exexcution-specs and the go library to understand this opcode.

I think that the go doc is worth mentioning, as it's much clearer than the exec-spec

https://pkg.go.dev/github.com/holiman/uint256#Int.ExtendSign